hummingbird.py

```diff
@@ -0,0 +1,153 @@
+		self.arrival = arrival
+
+	def has_not_played_today(self):
+		# not played today if:
+		# 	1) it's now after reset point and last played earlier today before reset point or earlier
+		# 	2) it's now before reset point and last played earlier yesterday before reset point or earlier
+		now = datetime.datetime.now()
+		today = now.date()
+		yesterday = today - datetime.timedelta(days=1)
+		return (
+			now.time() >= config.time_reset_time and self.arrival < datetime.datetime.combine(today, config.time_reset_time)
+		) or (
+			now.time() < config.time_reset_time and self.arrival < datetime.datetime.combine(yesterday, config.time_reset_time)
+		)
+
+	def queue_song(self, music_player):
+		self.arrival = datetime.datetime.now()
+		music_player.queue_song(user=self)
+
+	def __unicode__(self):
+		return "{name}: {song} ({length}s)".format(name=self.name, song=self.song, length=self.length)
+
+class System:
+	def __init__(self):
+		log(message="Initializing...")
+		# local_payload_cache uses mac_id ("address") as key, with the value being a dictionary of mac_id and datetime value of last time the api was pinged.
+		# Cache is only used if use_cache is set to True in config.py
+		self.local_payload_cache = {}		
+		self.music_player = MusicPlayer()
+		self.all_addresses = self.read_in_addresses()
+		self.waiting_for_input = False
+		self.input_timeout()
+		log(message="Waiting for tcpdump to provide input...")
+
+		while True:
+			# handle input as network traffic from tcpdump
+			addresses = get_MAC(line=stdin.readline())
+			self.waiting_for_input = False
+			for address in addresses:
+				
+				## Pings Hummingbird Django server. If use_cache is set to True, use self.local_payload_cache as a cache.
+				if config.use_cache:
+					if address.lower() in self.local_payload_cache:
+						# payload is a tuple of address and last_checked_datetime
+						payload = self.local_payload_cache[address.lower()]
+						# Set a 60 second cache				
+						if (datetime.datetime.now() - payload['last_sent_dt']).seconds > config.cache_time_seconds:
+							r = requests.get("http://127.0.0.1:8000/hummingbird/build_user_from_device/", params=payload)
+							payload['last_sent_dt'] = datetime.datetime.now()
+							self.local_payload_cache[address.lower()] = payload
+							cached = False
+						else:
+							cached = True
+					else:
+						payload = {'mac_id': address.lower(), 'last_sent_dt': datetime.datetime.now()}
+						self.local_payload_cache[address.lower()] = payload						
+						r = requests.get("http://127.0.0.1:8000/hummingbird/build_user_from_device/", params=payload)				
+						cached = False
+				else:
+					payload = {'mac_id': address.lower(), 'last_sent_dt': datetime.datetime.now()}
+					r = requests.get("http://127.0.0.1:8000/hummingbird/build_user_from_device/", params=payload)				
+					cached = False
+
+				## Convert the string representation of a dicitonary with user info into a dictionary object.
+				user_dict = ast.literal_eval(r.text)
```

Comment:
We should probably use a more standard JSON flow. In Django 1.7 you can generate a JSON response by returning a JsonResponse from the request, which takes in a dictionary.
Learn more here: https://docs.djangoproject.com/en/1.7/ref/request-response/#jsonresponse-objects

Then when receiving the request, the library `requests` can reformat this nicely as a dictionary with code that's something like:
```python
user_dict = requests.get("http://127.0.0.1:800/rest/of/uri").json()
```